core/livepeernode_test.go

```diff
@@ -84,90 +59,38 @@ func TestTranscodeAndBroadcast(t *testing.T) {
 		t.Errorf("Expecting %v profiles, got %v", len(p), len(res.Data))
 	}
 
-	//Should have transcoded the segments into 2 different profiles (right now StubSubscriber emits 1 segment)
-	if len(tr.InputData) != 1 {
-		t.Errorf("Expecting 1 segment to be transcoded, got %v", tr.InputData)
+	//Should have 1 transcoded segment into 2 different profiles
+	if tr.SegCount != 1 {
+		t.Errorf("Expecting 1 segment to be transcoded, got %v", tr.SegCount)
 	}
 
 	// Check playlist was updated
-	for _, v := range strmIds {
+	// XXX Fix this
```

Comment:
Should be fixed, but the test needs to be redone significantly to achieve the same level of coverage. The note is just so we don't forget.

Reply:
Ok.  Are we waiting for the next iteration to write this test?  Or should we do it now?

Reply:
Next iteration probably, since the playlist (OS actually) is updated in the RPC layer (`ServeSegment`) and that particular area is hard to test right now. There needs to be some refactoring for testability.

Most of these tests have been disabled since OS. We're gaining them back a little at a time though.